### PR TITLE
add permission edit button

### PR DESCRIPTION
**Pull request by sericaia**
Test Case
1. Permissions - add permission for app X
2. Roles - add role for app X
3. Roles - add permission created
4. Permissions - change app X to app Y
5. Roles - confirm that permission is no longer in permission options
6. Permissions - change app Y to app X again
7. Roles - permission is again on options, and it’s selected > this must happen?!
    - should it be unselected?
    - OR should it be possible to change application in permissions?

### CODE REVIEW

**Comment by sericaia** [comment]
Implementing "Should not allow changing the application of a permission if the permission is part of any role or group" requirement fixes the mentioned problem.

File: server/lib/storage/database.js [hunk 1]
```diff
@@ -36,13 +36,14 @@ export default class Database {
     return { size: null, type: config('DB_TYPE') };
   }
 
-  canDeleteRecord(type, checkFor, id) {
+  canTouchRecord(type, checkFor, id) {
+    console.log('type:', type, 'checkFor:', checkFor, 'id:', id)
```
Comment (sandrinodimattia) [inline server/lib/storage/database.js]: Remove this

**Comment by sandrinodimattia** [comment]
What is Touch? Do you mean Edit/Update?

File: server/lib/storage/database.js [hunk 2]
```diff
@@ -36,13 +36,13 @@ export default class Database {
     return { size: null, type: config('DB_TYPE') };
   }
 
-  canDeleteRecord(type, checkFor, id) {
+  canTouchRecord(type, checkFor, id) {
```
Comment (sericaia) [inline server/lib/storage/database.js]: it's to verify if we can make a change (update, delete, etc.)

Comment (sandrinodimattia) [inline server/lib/storage/database.js]: Please rename everything to "canChange" then

**Pull request merged**